pkg/pillar/cmd/vaultmgr/vaultmgr.go

```diff
@@ -0,0 +1,195 @@
+// Copyright (c) 2018-2019 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package vaultmgr
+
+import (
+	"bytes"
+	"flag"
+	"io/ioutil"
+	"os"
+	"os/exec"
+
+	"github.com/lf-edge/eve/pkg/pillar/agentlog"
+	"github.com/lf-edge/eve/pkg/pillar/cmd/tpmmgr"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	fscryptPath   = "/opt/zededa/bin/fscrypt"
+	keyctlPath    = "/bin/keyctl"
+	mountPoint    = "/persist/"
+	vaultPath     = "/persist/vault"
+	keyDir        = "/TmpVaultDir"
+	protectorName = "TheVaultProtector"
+	vaultKeyLen   = 32 //bytes
+)
+
+var (
+	keyFile        = keyDir + "/protector.key"
+	keyctlParams   = []string{"link", "@u", "@s"}
+	mntPointParams = []string{"setup", mountPoint, "--quiet"}
+	statusParams   = []string{"status", mountPoint}
+	setupParams    = []string{"setup", "--quiet"}
+	encryptParams  = []string{"encrypt", vaultPath, "--key=" + keyFile,
+		"--source=raw_key", "--name=" + protectorName,
+		"--user=root"}
+	unlockParams = []string{"unlock", vaultPath, "--key=" + keyFile,
+		"--user=root"}
+)
+
+func execCmd(command string, args ...string) (string, string, error) {
+	var stdout, stderr bytes.Buffer
+
+	cmd := exec.Command(command, args...)
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+
+	err := cmd.Run()
+	stdoutStr, stderrStr := stdout.String(), stderr.String()
+	return stdoutStr, stderrStr, err
+}
+
+func linkKeyrings() error {
+	if _, _, err := execCmd(keyctlPath, keyctlParams...); err != nil {
+		log.Fatalf("Error in linking user keyring %v", err)
+		return err
+	}
+	return nil
+}
+
+//stageKey is responsible for talking to TPM and Controller
+//and preparing the key for accessing the vault
+func stageKey() error {
+	//Create a tmpfs file to pass the secret to fscrypt
+	if _, _, err := execCmd("mkdir", keyDir); err != nil {
+		log.Fatalf("Error creating keyDir %v", err)
+		return err
+	}
+
+	if _, _, err := execCmd("mount", "-t", "tmpfs", "tmpfs", keyDir); err != nil {
+		log.Fatalf("Error mounting tmpfs on keyDir: %v", err)
+		return err
+	}
+
+	//XXX: This section will change once Controller integration is complete.
+	//XXX: For now assumes only TPM part of the key logic
```

Comment:
How much work will it be to implement the controller key part logic already? You can assume the key to be static for now (sine the controller part is not ready) and call it a 'foobar'. Still it would be useful to see the proposed logic of combining TPM and controller key material in this implementation ASAP.

Reply:
I've added merge logic with a dummy cloudKey. Please have a look